mgmt/rest/rest_v2_test.go

```diff
@@ -101,29 +134,209 @@ func startV2API(cfg *mockConfig, testType string) *restAPIInstance {
 	}
 }
 
-func TestV2Metric(t *testing.T) {
-	r := startV2API(getDefaultMockConfig(), "metric")
-	Convey("Test Metric REST API V2", t, func() {
-		Convey("Get metrics - v2/metrics", func() {
+func TestV2Plugin(t *testing.T) {
+	r := startV2API(getDefaultMockConfig(), "plugin")
+	Convey("Test Plugin REST API V2", t, func() {
+
+		Convey("Post plugins - v2/plugins/:type:name", func(c C) {
+			f, err := os.Open(MOCK_PLUGIN_PATH1)
+			defer f.Close()
+			So(err, ShouldBeNil)
+
+			// We create a pipe so that we can write the file in multipart
+			// format and read it in to the body of the http request
+			reader, writer := io.Pipe()
+			mwriter := multipart.NewWriter(writer)
+			bufin := bufio.NewReader(f)
+
+			// A go routine is needed since we must write the multipart file
+			// to the pipe so we can read from it in the http call
+			go func() {
+				part, err := mwriter.CreateFormFile("snap-plugins", "mock")
+				c.So(err, ShouldBeNil)
+				bufin.WriteTo(part)
+				mwriter.Close()
+				writer.Close()
+			}()
+
+			resp1, err1 := http.Post(
```

Comment:
From the comments in the file,  ` A go routine is needed since we must write the multipart file to the pipe so we can read from it in the http call ` 
I agree with @kindermoumoute, I don't recall ever seeing this fail.